analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ClearTextProtocolsAreSensitive.cs

```diff
@@ -134,10 +134,32 @@ private void VisitStringExpressions(SyntaxNodeAnalysisContext c)
             objectCreation.ArgumentList?.Arguments.Count > 0
             && validServerRegex.IsMatch(GetText(objectCreation.ArgumentList.Arguments[0].Expression));
 
-        private string GetUnsafeProtocol(string text)
+        private string GetUnsafeProtocol(string text, SyntaxNode node)
         {
             if (httpRegex.IsMatch(text))
             {
+                switch (node.Parent)
```

Comment:
this looks like it could be a method "isNamespace", arrow-bodies with a switch expression

and then the `if` condition on line 139 would be `if (httpRegex.IsMatch(text) && !isNamespace())`